sdk/src/frame_impl.cpp

```diff
@@ -97,30 +93,36 @@ aditof::Status FrameImpl::getDetails(aditof::FrameDetails &details) const {
     return aditof::Status::OK;
 }
 
-aditof::Status FrameImpl::getData(aditof::FrameDataType dataType,
+aditof::Status FrameImpl::getData(const std::string &dataType,
                                   uint16_t **dataPtr) {
     using namespace aditof;
 
-    switch (dataType) {
-    case FrameDataType::FULL_DATA: {
-        *dataPtr = m_fullData;
-        break;
-    }
-    case FrameDataType::IR: {
-        *dataPtr = m_irData;
-        break;
-    }
-    case FrameDataType::DEPTH: {
-        *dataPtr = m_depthData;
-        break;
-    }
+    if (m_dataLocations.count(dataType) > 0) {
+        *dataPtr = m_dataLocations[dataType];
+    } else {
+        dataPtr = nullptr;
+        LOG(ERROR) << dataType << " is not supported by this frame!";
+        return Status::INVALID_ARGUMENT;
     }
 
     return Status::OK;
 }
 
 void FrameImpl::allocFrameData(const aditof::FrameDetails &details) {
-    m_fullData = new uint16_t[details.width * details.height * 2];
-    m_depthData = m_fullData;
-    m_irData = m_fullData + (details.width * details.height);
+    unsigned int totalWidth = 0;
+    unsigned int totalHeigth = 0;
+
+    for (const auto &details : details.dataDetails) {
+        totalWidth += details.width;
+        totalHeigth += details.height;
+    }
+
+    m_allData = new uint16_t[totalWidth * totalHeigth];
```

Comment:
why not use `details.width` and `details.height`?

Reply:
Actually this is where details.width and details.height should be calculated. However, I don't think we need them at all (outside FrameImpl).
I don't think we will be adding metadata for this camera. For other cameras, they will have their own FrameImpl.